model/task/db.go

```diff
@@ -812,6 +812,63 @@ func GetRecentTaskStats(period time.Duration, nameKey string) ([]StatusItem, err
 	return result, nil
 }
 
+// ByExecutionTasksAndMaxExecution returns the tasks corresponding to the passed in taskIds and execution,
+// or the most recent executions of those tasks if they do not have a matching execution
+func ByExecutionTasksAndMaxExecution(taskIds []*string, execution int) ([]Task, error) {
```

Comment:
Can you call this FindBy... ? Typically when our DB functions are just By, they only return the query and then they can be used with Find, Count, whatever, but this includes the Find inside the function.